### PR TITLE
Create initial cachex config

### PR DESCRIPTION
**Pull request by faultables**
ni konfigurasi yg dipakai di cache.komunix.org

### CODE REVIEW

File: cachex/configuration.nix [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            middlewares = [ "cachex_index" "cachex_fallback" ];
+          };
+
+          cachex_index = {
+            rule = "Host(`${cachexDomain}`) && Path(`/`)";
+            service = "cachex_index";
+            priority = 1337;
+            middlewares = [ "cachex_index" ];
+          };
+        };
```
Comment (faultables) [inline cachex/configuration.nix]: actually we don't need this anymore as I use nginx heavily

File: cachex/configuration.nix [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            { url = "http://127.0.0.1:8080/"; }
+          ];
+
+          cachex_fallback.loadBalancer.servers = [
+            { url = "http://127.0.0.1:8080/"; }
+          ];
+
+          cachex_index.loadBalancer.servers = [
+            { url = "http://127.0.0.1:2022/"; }
+          ];
```
Comment (faultables) [inline cachex/configuration.nix]: idem

File: cachex/configuration.nix [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+          cachex_fallback = {
+            headers.customResponseHeaders."X-Komunix-Fallback-To" = "${fallbackUpstream}";
+          };
+        };
+
+        routers = {
+          cachex = {
+            rule = "Host(`${cachexDomain}`) && PathPrefix(`/`)";
+            service = "cachex_fallback";
+            priority = 1;
```
Comment (faultables) [inline cachex/configuration.nix]:
```suggestion

```

File: cachex/configuration.nix [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  environment.systemPackages = with pkgs; [
+    vim
+    vnstat
+    wget
+    lsof
+  ];
+
+  services.openssh.enable = true;
+
+  services.nginx = {
```
Comment (faultables) [inline cachex/configuration.nix:40]: maybe use openresty?

**Comment by faultables** [comment]
gw ada pertanyaan. given di repo ini ada direktori `cachex/cachex_index.html`, gw kurang tau gimana caranya si file tersebut:

- dipindahkan ke `/srv/www/cachex_index/`
- diubah namanya menjadi `index.html`
- hasil akhirnya adalah: `./cachex/cachex_index.html` -> `/srv/www/cachex_index/index.html`

secara ✨ deklaratif ✨

---

sebagai solusi sementara, gw ngelakuinnya secara ~~imperatif~~ manual dengan:

```bash
mkdir -p /srv/www/cachex_index; \
  chown -R nginx:nginx /srv/www/cachex_index; \
  cp ./cachex_index.html /srv/www/cachex_index/index.html
```

you know sysadmins thing.

would be great klo bisa dilakukan secara declarative biar reProduCibLe

/cc orang gila @r17x

**Comment by r17x** [comment]
@faultables  this is [what you need](https://github.com/NixOS/nixpkgs/blob/2ddc335e6f32b875e14ad9610101325b306a0add/nixos/modules/system/activation/activation-script.nix#L214-L228) to NOT DOING manual script.

**Comment by r17x** [comment]
Another References:

* https://danilafe.com/blog/blog_with_nix/
* https://nixos.org/manual/nixpkgs/stable/#sec-getting-documentation
image

**Comment by faultables** [comment]
gw dapet cara yang lebih elegan lol https://github.com/KomuNix/503/pull/1/commits/509017bb93da80340f687eeadb669d078183a4f0

image

File: cachex/configuration.nix (hunk 4 above)
Comment (faultables) [reply cachex/configuration.nix:40]: nanti aja deh mager

**Comment by faultables** [comment]
bentar one more commit

**Comment by faultables** [comment]
rede i guess

**Review by r17x — APPROVED** [review]
LgTM Redeeee

**Comment by r17x** [comment]
LGTM POST MERGED

**Pull request merged**